fs/io_zenfs.cc

```diff
@@ -215,6 +240,13 @@ Status ZoneFile::MergeUpdate(std::shared_ptr<ZoneFile> update, bool replace) {
   is_sparse_ = update->IsSparse();
   MetadataSynced();
 
+  if (nr_links_ < update->GetNrLinks()) {
```

Comment:
Is the check correct, that a ZoneFile update can only contain more links? Could it not happen that a ZoneFile update contains less links and thus we need to remove links that do not exist from `this` ZoneFile?

Reply:
Yes it is correct.
When a link is added we encode it as an Update, so when decoding updates, we do a  merging of new updates, so we will be adding a new updated link information to the current snapshot. 

When a link is deleted it is encoded as a FileLinkDeletion operation, it gets decoded at DecodeFileLinkDeletionFrom() , here the file is removed from files_ map and  linkfiles_ vector and links counts are accounted.